apollo-router/src/router_factory.rs

```diff
@@ -1,113 +1,45 @@
 use crate::apollo_router::{ApolloPreparedQuery, ApolloRouter};
 use crate::configuration::Configuration;
 use crate::http_service_registry::HttpServiceRegistry;
-use apollo_router_core::prelude::{graphql::*, *};
-use futures::prelude::*;
+use apollo_router_core::prelude::*;
 use std::sync::Arc;
 
 /// Factory for creating graphs.
 ///
 /// This trait enables us to test that `StateMachine` correctly recreates the ApolloRouter when
 /// necessary e.g. when schema changes.
-//#[cfg_attr(test, automock)]
+#[async_trait::async_trait]
 pub(crate) trait RouterFactory<Router, PreparedQuery>
 where
     Router: graphql::Router<PreparedQuery>,
     PreparedQuery: graphql::PreparedQuery,
 {
-    fn create(
+    async fn create(
         &self,
         configuration: &Configuration,
         schema: Arc<graphql::Schema>,
-        plan_cache_limit: usize,
-        query_cache_limit: usize,
-    ) -> future::BoxFuture<'static, Router>;
-    fn recreate(
-        &self,
-        router: Arc<Router>,
-        configuration: &Configuration,
-        schema: Arc<graphql::Schema>,
-        plan_cache_limit: usize,
-        query_cache_limit: usize,
-    ) -> future::BoxFuture<'static, Router>;
-    fn get_plan_cache_limit(&self) -> usize;
-    fn get_query_cache_limit(&self) -> usize;
+        previous_router: Option<Arc<Router>>,
+    ) -> Router;
 }
 
 #[derive(Default)]
-pub(crate) struct ApolloRouterFactory {
-    plan_cache_limit: usize,
-    query_cache_limit: usize,
-}
+pub(crate) struct ApolloRouterFactory {}
+
 impl ApolloRouterFactory {
-    pub fn new(plan_cache_limit: usize, query_cache_limit: usize) -> Self {
-        Self {
-            plan_cache_limit,
-            query_cache_limit,
-        }
+    pub fn new() -> Self {
```

Comment:
this looks a lot like Default, which is derived above. Do we have something in mind?

Reply:
I think you are right. I added new() to cover the parameters. Now they have gone, I think we could remove new() and just use default().

Reply:
Fixed in f218f49